sound/soc/sof/ipc4-pcm.c

```diff
@@ -106,7 +211,29 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 		pipeline->state = state;
 	}
 
-	return ret;
+	if (ret < 0)
+		return ret;
+
+	if (state == SOF_IPC4_PIPE_RUNNING) {
+		u32 abi_version;
+		u32 abi_offset;
+
+		abi_offset = offsetof(struct sof_ipc4_fw_registers, abi_ver);
+		sof_mailbox_read(sdev, sdev->fw_info_box.offset + abi_offset, &abi_version,
+				 sizeof(abi_version));
+
+		if (abi_version != SOF_IPC4_FW_REGS_ABI_VER) {
+			dev_warn(sdev->dev, "unsupported abi version %d for time stamp\n",
+				 abi_version);
+			return -EINVAL;
+		}
```

Comment:
I am still very very confused about this ABI check. @kv2019i @lgirdwood can you confirm this is the right approach? I have lost track of IPC4 ABI threads.

Reply:
@plbossart I think this is ok. It's a bit confusing as this an independent and local versioning scheme for this specific portion of the mapped SRAM area, and not related to the overall SOF ABI version. But I think this is ok defined like this. The documentation in the PR does state this is the version of the struct specifically.